docs/ops/movement/GatherTree_1.md

```diff
@@ -2,63 +2,60 @@
 
-The GatherTree operation implements the same algorithm as the [GatherTree operation in TensorFlow](https://www.tensorflow.org/addons/api_docs/python/tfa/seq2seq/gather_tree).
+*GatherTree* operation reorders token IDs of a given input tensor `step_id` representing IDs per each step of beam search, based on input tensor `parent_ids` representing the parent beam IDs. For a given beam, past the time step containing the first decoded `end_token` all values are filled in with end_token.
 
-Pseudo code:
+The algorithm in pseudocode is as follows:
 
 ```python
-final_idx[ :, :, :] = end_token
+final_ids[ :, :, :] = end_token
 for batch in range(BATCH_SIZE):
     for beam in range(BEAM_WIDTH):
         max_sequence_in_beam = min(MAX_TIME, max_seq_len[batch])
 
-        parent = parent_idx[max_sequence_in_beam - 1, batch, beam]
+        parent = parent_ids[max_sequence_in_beam - 1, batch, beam]
 
-        final_idx[max_sequence_in_beam - 1, batch, beam] = step_idx[max_sequence_in_beam - 1, batch, beam]
+        final_ids[max_sequence_in_beam - 1, batch, beam] = step_ids[max_sequence_in_beam - 1, batch, beam]
 
         for level in reversed(range(max_sequence_in_beam - 1)):
-            final_idx[level, batch, beam] = step_idx[level, batch, parent]
+            final_ids[level, batch, beam] = step_ids[level, batch, parent]
 
-            parent = parent_idx[level, batch, parent]
+            parent = parent_ids[level, batch, parent]
 
         # For a given beam, past the time step containing the first decoded end_token
         # all values are filled in with end_token.
         finished = False
         for time in range(max_sequence_in_beam):
             if(finished):
-                final_idx[time, batch, beam] = end_token
-            elif(final_idx[time, batch, beam] == end_token):
+                final_ids[time, batch, beam] = end_token
+            elif(final_ids[time, batch, beam] == end_token):
                 finished = True
 ```
 
-Element data types for all input tensors should match each other.
+*GatherTree* operation is equivalent to [GatherTree operation in TensorFlow](https://www.tensorflow.org/addons/api_docs/python/tfa/seq2seq/gather_tree).
 
-**Attributes**: *GatherTree* has no attributes
+**Attributes**: *GatherTree* operation has no attributes.
 
 **Inputs**
 
-* **1**:  `step_ids` -- a tensor of shape `[MAX_TIME, BATCH_SIZE, BEAM_WIDTH]` of type *T* with indices from per each step. **Required.**
-
-* **2**:  `parent_idx` -- a tensor of shape `[MAX_TIME, BATCH_SIZE, BEAM_WIDTH]` of type *T* with parent beam indices. **Required.**
-
-* **3**:  `max_seq_len` -- a tensor of shape `[BATCH_SIZE]` of type *T* with maximum lengths for each sequence in the batch. **Required.**
-
-* **4**:  `end_token` -- a scalar tensor of type *T* with value of the end marker in a sequence. **Required.**
-
+* **1**:  `step_ids` - Indices from per each step. A tensor of type *T* and rank 3. Layout is `[MAX_TIME, BATCH_SIZE, BEAM_WIDTH]`. **Required.**
```

Comment:
I think we could make it a bit more clear here in the first sentence.

Reply:
done